apis/v1alpha2/httproute_types.go

```diff
@@ -212,11 +212,11 @@ type HTTPRouteRule struct {
 	// * a resource in another namespace when the reference has not been
 	//   explicitly allowed by a ReferenceGrant (or equivalent concept).
 	//
-	// When a BackendRef is invalid, 404 status codes MUST be returned for
+	// When a BackendRef is invalid, 500 status codes MUST be returned for
 	// requests that would have otherwise been routed to an invalid backend. If
 	// multiple backends are specified, and some are invalid, the proportion of
 	// requests that would otherwise have been routed to an invalid backend
-	// MUST receive a 404 status code.
+	// MUST receive a 500 status code.
 	//
 	// When a BackendRef refers to a Service that has no ready endpoints, it is
 	// recommended to return a 503 status code.
```

Comment:
Should this have been changed to 500 too?

Reply:
I agree, seems we should update here as well.
@robscott should I make the PR to change this status code 503 -> 500 as well?

Reply:
Added the PR for that: https://github.com/kubernetes-sigs/gateway-api/pull/1210